environs/bootstrap/bootstrap_test.go

```diff
@@ -931,6 +934,106 @@ func (s *bootstrapSuite) TestBootstrapMetadata(c *gc.C) {
 	c.Assert(env.instanceConfig.Bootstrap.CustomImageMetadata[0], gc.DeepEquals, metadata[0])
 }
 
+func (s *bootstrapSuite) TestBootstrapMetadataDirNonexistend(c *gc.C) {
+	env := newEnviron("foo", useDefaultKeys, nil)
+	nonExistentFileName := "/tmp/TestBootstrapMetadataDirNonexistend"
+	err := bootstrap.Bootstrap(envtesting.BootstrapContext(c), env, bootstrap.BootstrapParams{
+		ControllerConfig: coretesting.FakeControllerConfig(),
+		AdminSecret:      "admin-secret",
+		CAPrivateKey:     coretesting.CAKey,
+		MetadataDir:      nonExistentFileName,
+	})
+	c.Assert(err, gc.NotNil)
+	c.Assert(err, gc.ErrorMatches, fmt.Sprintf("simplestreams metadata source: %s not found", nonExistentFileName))
+}
+
+// TestBootstrapMetadataImagesNoTools tests 2 cases:
+// juju bootstrap --metadata-source <dir>
+// juju bootstrap --metadata-source <dir>/images
+// where <dir>/tools doesn't exist
+func (s *bootstrapSuite) TestBootstrapMetadataImagesNoTools(c *gc.C) {
```

Comment:
this should either be 2 tests, or a single test inside a for loop
metadataDir, _ := createImageMetadata(c)
for _, suffix := range []string{"", "images"} {
  sourceDir := filepath.Join(metadataDir, suffix)
  do test stuff...
}